src/sonic-host-services/scripts/hostcfgd

```diff
@@ -136,8 +158,17 @@ class FeatureHandler(object):
         # the next called self.update_feature_state will start it again. If it will fail
         # again the auto restart will kick-in. Another order may leave it in failed state
         # and not auto restart.
-        if self._cached_config[feature_name].auto_restart != feature.auto_restart:
+
+        dir_name = self.SYSTEMD_SERVICE_CONF_DIR.format(feature_name)
+        auto_restart_conf = os.path.join(dir_name, 'auto_restart.conf')
+        updated_auto_restart = False
+        if not os.path.exists(auto_restart_conf):  # if the auto_restart_conf file is not found, set it
```

Comment:
What if 'auto_restart.conf' is found but does not have a correct value?

Reply:
In that case, this condition will kick-in to modify the conf file. 

```
if self._cached_config[feature_name].auto_restart != feature.auto_restart:
            if not updated_auto_restart:
                self.update_feature_auto_restart(feature)
            self._cached_config[feature_name].auto_restart = feature.auto_restart 
```
I'm assuming the potential state change, self._cached_config[feature_name].auto_restart was set to a wrong value and a new notification was received with a correct value. Let me know if you have another scenario in mind

Reply:
Manual modification of auto_restart.conf by hand.

Reply:
No, such scenario was not handled before and am not covering it now.